src/OtcOption.sol

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+error Option__InvalidUnderlyingToken();
+error Option__InvalidQuoteToken();
+error Option__InvalidStrike();
+error Option__InvalidMaturity();
+error Option__InvalidAmount();
+error Option__InvalidPremium();
+
+error Option__PriceFeedNotFound();
+error Option__DepositPremiumFailed();
+error Option__DepositMarginFailed();
+error Option__TransferPremiumFailed();
+error Option__TransferMarginFailed();
+error Option__DealNotOpen();
+error Option__DealExpired();
+error Option__DealNotExpired();
+error Option__DealNotFound();
+error Option__DealHasNoTaker();
+
+contract OtcOption is Ownable {
+    event DealCreated(
+        uint id,
+        address underlyingToken,
+        address quoteToken,
+        uint strike,
+        uint maturity,
+        bool isCall,
+        uint amount,
+        uint premium,
+        bool isMakerBuyer,
+        address maker
+    );
+    event DealTaken(uint id, address taker);
+    event DealRemoved(uint id);
+    event DealSettled(uint id, bool exercised);
+
+    enum DealStatus {
+        Open,
+        Taken,
+        Removed,
+        Settled
+    }
+
+    struct Deal {
+        uint id;
+        address underlyingToken;
+        address quoteToken;
+        uint strike;
+        uint maturity;
+        bool isCall;
```

Comment:
Need to check all the correct uint type. As we can rearrange the struct and optimize gas